engines/toon/movie.cpp

```diff
@@ -115,9 +119,30 @@ void Movie::playVideo(bool isFirstIntroVideo) {
 				} else {
 					_vm->_system->copyRectToScreen(frame->getPixels(), frame->pitch, 0, 0, frame->w, frame->h);
 
+					int32 currentFrame = _decoder->getCurFrame();
+
+					// find unused color key to replace with subtitles color
+					int len = frame->w * frame->h;
+					const byte* pixels = (const byte *)frame->getPixels();
+					byte counts[256];
+					memset(counts, 0, sizeof(counts));
+					for (int i = 0; i < len; i++) {
+						counts[pixels[i]] = 1;
+					}
+
+					// 0 is already used for the border color and should not be used here, so it can be skipped over.
+					for (int j = 1; j < 256; j++) {
+						if (counts[j] == 0) {
+							unused = j;
+							break;
+						}
+					}
+
+					_subtitle->render(*frame, currentFrame, unused);
+
 					// WORKAROUND: There is an encoding glitch in the first intro video. This hides this using the adjacent pixels.
 					if (isFirstIntroVideo) {
-						int32 currentFrame = _decoder->getCurFrame();
+						// int32 currentFrame = _decoder->getCurFrame();
```

Comment:
Why this is commented out now? If you're referring to the new call above, then don't you change the semantics, as this call was only when `isFirstIntroVideo` is set?

Reply:
(sorry, I don't think I understand the question) this line was moved (line 122) so it's now unconditional as frame information should now be used for all videos and not just this workaround here.